Mathlib/Geometry/RingedSpace/Basic.lean

```diff
@@ -108,21 +135,28 @@ theorem isUnit_of_isUnit_germ (U : Opens X) (f : X.presheaf.obj (op U))
     rw [germ_res_apply, germ_res_apply]
     apply (h z ((iVU x).le hzVx)).mul_right_inj.mp
     -- Porting note: now need explicitly typing the rewrites
-    rw [← X.presheaf.germ_res_apply (iVU x) z hzVx f]
+    -- note: this is bad, I think we should replace the `FunLike` on
+    -- concrete category with `CoeFun`
```

Comment:
I think we should see it as separating the "concrete category is one with a forgetful functor to `Type`" and "concrete category is one where the morphisms are `FunLike`s". The latter can imply the former, but crucially we'd need the actual `FunLike` instances to be passed in, not created by `ConcreteCategory`.

Reply:
Agreed that we shouldn't put a `FunLike` on `X ⟶ Y`. Rather, like we already did we could have a `CoeFun` instance that returns some type `F` that does have a `FunLike` instance.